File: 1-25/euler001.cpp
#include <iostream>
#define LIMIT (1000-1)

int main()
{
	int sum = 0;
	sum += 3 * (LIMIT/3+1) * (LIMIT/3) / 2;
	sum += 5 * (LIMIT/5+1) * (LIMIT/5) / 2;
	sum -= 15 * (LIMIT/15+1) * (LIMIT/15) / 2;
	std::cout << sum << "\n";
}

File: 1-25/euler001.rs
fn sum(n: i32, accum: i32) -> i32 {
    if n <= 0 { accum }
    else if n % 3 == 0 { sum(n-1, accum+n) }
    else if n % 5 == 0 { sum(n-1, accum+n) }
    else { sum(n-1, accum) }
}

fn main() {
    println!("{}", sum(999, 0));
}
